eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -27,62 +28,80 @@ parameters:
-            Write-Host "Creating Resource Group: $($RgName)"
-            Write-Host "##vso[task.setvariable variable=AZURE_RESOURCEGROUP_NAME]$RgName"
+        - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
+          - pwsh: |
+              $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+              $RgName = "rg-${{ parameters.ServiceDirectory }}-$($BaseName)"
+              $ResourceGroupExpiration = (Get-Date).ToUniversalTime().AddHours(24).ToString("o")
+              Write-Host "Creating Resource Group: $($RgName)"
+              Write-Host "##vso[task.setvariable variable=AZURE_RESOURCEGROUP_NAME]$RgName"
 
-            az login --service-principal -u $(provisioner-aad-id) -p $(provisioner-aad-secret) --tenant $(provisioner-aad-tenant)
-            az account set --subscription $(provisioner-subscription)
+              az login --service-principal -u $(provisioner-aad-id) -p $(provisioner-aad-secret) --tenant $(provisioner-aad-tenant)
+              az account set --subscription $(provisioner-subscription)
 
-            az group create --location westus2 --name $RgName --tags DeleteAfter=$ResourceGroupExpiration BuildId='$(Build.BuildId)' BuildJob='$(Agent.JobName)' BuildNumber='$(Build.BuildNumber)' BuildReason='$(Build.Reason)'
-            az group wait --created --resource-group $RgName
-          displayName: Create resource group and environment variable for tests
+              az group create --location westus2 --name $RgName --tags DeleteAfter=$ResourceGroupExpiration BuildId='$(Build.BuildId)' BuildJob='$(Agent.JobName)' BuildNumber='$(Build.BuildNumber)' BuildReason='$(Build.Reason)'
+              az group wait --created --resource-group $RgName
+            displayName: Create resource group and environment variable for tests
 
-      - template: ../steps/build-test.yml
-        parameters:
-          BuildTargetingString: ${{ parameters.BuildTargetingString }}
-          ServiceDirectory: ${{ parameters.ServiceDirectory }}
-          CoverageArg: $(CoverageArg)
-          EnvVars:
-            AZURE_RUN_MODE: 'Live' #Record, Playback
-            ${{ insert }}: ${{ parameters.EnvVars }}
-          BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
-          AfterTestSteps: ${{ parameters.AfterTestSteps }}
-          PythonVersion: $(PythonVersion)
-          OSVmImage: $(OSVmImage)
-          ToxTestEnv: "whl"
-          AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
-          TestMarkArgument: ${{ parameters.TestMarkArgument }}
-          InjectedPackages: ${{ parameters.InjectedPackages }}
-          BuildDocs: ${{parameters.BuildDocs}}
+          # figure out what's up with the create resource group above
+        - ${{ if eq(parameters['AllocateResourceGroup'], 'false') }}:
+          - template: ../../../common/TestResources/deploy-test-resources.yml
+            parameters:
+              Location: ${{ coalesce(parameters.Location, cloudConfig.value.Location, 'westus2') }}
```

Comment:
I would remove the 'westus2' default here (similar to https://github.com/Azure/azure-sdk-for-net/blob/master/eng/pipelines/templates/stages/archetype-sdk-tests.yml#L140). We had a bug with this original expression because it sets the wrong region default for non-public clouds.